### PR TITLE
Car 69 landing page

### PR DESCRIPTION
**Pull request by ElizavetaBuraya**

### CODE REVIEW

**Review by synthetiquely — APPROVED** [review]

**Review by BobriK502 — APPROVED** [review]

File: spec/landingPage.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  };
+
+  it("should throw validation error if name or email don't pass validation rules", async () => {
+    try {
+      await chai
+        .request(app)
+        .post('/api/user/send-message')
+        .set('content-type', 'application/json')
+        .send({ email: 'invalid', name: '123' });
+      assert.fail('Test failed. No validation error thrown for wrong username and email format');
```
Comment (AntonNarkevich) [inline spec/landingPage.ts:27]: This looks amazing! Now we have readable tests :)

File: src/services/userService.ts [hunk 2]
```diff
@@ -12,6 +13,14 @@ export const register = async (payload: IUser) => {
   }
 };
 
+export const sendMessage = async (data: IMessage) => {
+  try {
+    await sendUserEmail(data, emailActions.SEND_USER_MESSAGE);
+  } catch (error) {
```
Comment (AntonNarkevich) [inline src/services/userService.ts]: Do we really need this try-catch block?

File: src/utils/userMessage.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    const subject = getEmailSubject(action);
+    transport.sendMail(
+      {
+        from: data.email,
+        html,
+        subject,
+        to: mail.auth.user
+      },
+      (err, info) => {
+        global.console.log(`The mail sent to ${mail.auth.user}`);
```
Comment (AntonNarkevich) [inline src/utils/userMessage.ts]: Wait, why are we using global like that?

File: src/utils/userMessage.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import nodemailer = require('nodemailer');
+import { mail } from '../config/config';
+import { IMessage } from '../interfaces/index';
+import { getEmailSubject } from './index';
+
+const transport = nodemailer.createTransport(mail);
+
+export const sendUserEmail = (data: IMessage, action: string): void => {
+  if (process.env.NODE_ENV !== 'test') {
```
Comment (AntonNarkevich) [inline src/utils/userMessage.ts]: Please do if (wrong) {return;} it makes the code readable.

File: src/utils/userMessage.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import nodemailer = require('nodemailer');
+import { mail } from '../config/config';
+import { IMessage } from '../interfaces/index';
+import { getEmailSubject } from './index';
+
+const transport = nodemailer.createTransport(mail);
+
+export const sendUserEmail = (data: IMessage, action: string): void => {
```
Comment (AntonNarkevich) [inline src/utils/userMessage.ts]: This function needs to be async.

File: src/utils/userMessage.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      {
+        from: data.email,
+        html,
+        subject,
+        to: mail.auth.user
+      },
+      (err, info) => {
+        global.console.log(`The mail sent to ${mail.auth.user}`);
+        if (err) {
+          throw err;
```
Comment (AntonNarkevich) [inline src/utils/userMessage.ts]: This throw won't work. It happens out of context.

**Review by AntonNarkevich — CHANGES_REQUESTED** [review]
Please fix some issues I commented.
The rest looks really good.

File: src/utils/userMessage.ts [hunk 7]
```diff
@@ -5,25 +5,22 @@ import { getEmailSubject } from './index';
...
-        }
-      }
-    );
+export const sendUserEmail = async (data: IMessage, action: string) => {
+  if (process.env.NODE_ENV === 'test') {
+    return;
   }
+  const html = generateEmail(data.name, data.email, data.message);
+  const subject = getEmailSubject(action);
+  transport
```
Comment (AntonNarkevich) [inline src/utils/userMessage.ts]: Why don't we do 'await' here?

File: src/utils/userMessage.ts [hunk 8]
```diff
@@ -5,25 +5,22 @@ import { getEmailSubject } from './index';
...
+    .sendMail({
+      from: data.email,
+      html,
+      subject,
+      to: mail.auth.user
+    })
+    .catch(error => {
```
Comment (AntonNarkevich) [inline src/utils/userMessage.ts]: If we'd do await above we wouldn't need this catch at all.

File: src/utils/userMessage.ts [hunk 9]
```diff
@@ -5,25 +5,22 @@ import { getEmailSubject } from './index';
...
+      throw error;
```
Comment (AntonNarkevich) [inline src/utils/userMessage.ts]: Throwing in catch block.... looks a bit odd.

**Review by AntonNarkevich — CHANGES_REQUESTED** [review]

**Pull request merged**